pkg/controller/template/template_controller.go

```diff
@@ -171,12 +182,55 @@ func (ctrl *Controller) deleteSecret(obj interface{}) {
 	}
 }
 
+func (ctrl *Controller) enqueueController() {
+	cfg, err := ctrl.ccLister.Get(ctrlcommon.ControllerConfigName)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("couldn't get ControllerConfig on dependency callback %#v", err))
```

Comment:
the only thing is that changing this string makes the error much more generic

Reply:
Could pass the type into this function as a string and format string it into the error? Bit of an odd pattern though maybe?